wallet/src/de/schildbach/wallet/WalletApplication.java

```diff
@@ -749,9 +752,7 @@ public void stopBlockchainService() {
     }
 
     public void resetBlockchainState() {
-        Executors.newSingleThreadExecutor().execute(() -> {
-            blockchainStateDao.save(new BlockchainState(true));
-        });
+        blockchainStateDataProvider.resetBlockchainState();
```

Comment:
Moved this function to BlockchainStateDataProvider to run in a coroutine -- same for the next change.